src/components/charts/cdfChartD3.js

```diff
@@ -0,0 +1,405 @@
+    return this;
+  }
+
+  marginLeft(marginLeft) {
+    this.attrs.marginLeft = marginLeft;
+    return this;
+  }
+
+  marginRight(marginRight) {
+    this.attrs.marginRight = marginRight;
+    return this;
+  }
+
+  marginTop(marginTop) {
+    this.attrs.marginTop = marginTop;
+    return this;
+  }
+
+  showDistributionLines(showDistributionLines) {
+    this.attrs.showDistributionLines = showDistributionLines;
+    return this;
+  }
+
+  verticalLine(verticalLine) {
+    this.attrs.verticalLine = verticalLine;
+    return this;
+  }
+
+  showVerticalLine(showVerticalLine) {
+    this.attrs.showVerticalLine = showVerticalLine;
+    return this;
+  }
+
+  container(container) {
+    this.attrs.container = container;
+    return this;
+  }
+
+  data(data) {
+    this.attrs.data = data;
+    return this;
+  }
+
+  /**
+   * @param key
+   * @returns {[]}
+   */
+  getDataPoints(key) {
+    const dt = [];
+    const data = this.attrs.data[key];
+    const len = data.xs.length;
+
+    for (let i = 0; i < len; i++) {
+      dt.push({ x: data.xs[i], y: data.ys[i] });
+    }
+
+    return dt;
+  }
+
+  render() {
+    const attrs = this.attrs;
+    const container = d3.select(attrs.container);
+    if (container.node() === null) {
+      console.error('Container for D3 is not defined.');
+      return;
+    }
+
+    // Sets the width from the DOM element.
+    const containerRect = container.node().getBoundingClientRect();
+    if (containerRect.width > 0) {
+      attrs.svgWidth = containerRect.width;
+    }
+
+    // Calculated properties.
+    const calc = {};
+    calc.chartLeftMargin = attrs.marginLeft;
+    calc.chartTopMargin = attrs.marginTop;
+    calc.chartWidth = attrs.svgWidth - attrs.marginRight - attrs.marginLeft;
+    calc.chartHeight = attrs.svgHeight - attrs.marginBottom - attrs.marginTop;
+
+    const areaColorRange = d3.scaleOrdinal().range(attrs.areaColors);
+    this.dataPoints = [this.getDataPoints('primary')];
+
+    // Scales.
+    const xMin = d3.min(attrs.data.primary.xs);
+    const xMax = d3.max(attrs.data.primary.xs);
+
+    if (attrs.scale === 'linear') {
+      this.xScale = d3.scaleLinear()
+        .domain([attrs.minX || xMin, attrs.maxX || xMax])
+        .range([0, calc.chartWidth]);
+    } else {
+      this.xScale = d3.scaleLog()
+        .base(attrs.logBase)
+        .domain([attrs.minX, attrs.maxX])
+        .range([0, calc.chartWidth]);
+    }
+
+    const yMin = d3.min(attrs.data.primary.ys);
+    const yMax = d3.max(attrs.data.primary.ys);
+
+    this.yScale = d3.scaleLinear()
+      .domain([yMin, yMax])
+      .range([calc.chartHeight, 0]);
+
+    // Axis generator.
+    if (!!this.attrs.timeScale) {
+      const zero = _.get(this.attrs.timeScale, 'zero', moment());
+      const step = _.get(this.attrs.timeScale, 'step', 'years');
+      const length = _.get(this.attrs.timeScale, 'length', moment());
+
+      const xScaleTime = d3.scaleTime()
+        .domain([zero.toDate(), length.toDate()])
```

Comment:
I think this isn't quite right. 

If a distribution sets zero to now, and has a unit of 1 year, then a normal distribution from -1 to 1 should show the time domain of negative 1 year to + 1 year. It would be better to have this be automatic, or we can have it passed in at a higher level (where we can use reasonML to find wide percentiles, and do the date concersions).

Plus, we'd probably want to be able to pass in the domain bounds from the outer ReasonML component anyway.

Reply:
Oh, got you, you mean that a one as a unit (1) is equal to one year in this case? Will fix. Thanks.

Reply:
Yes, if the the unit=`year. If the time Vector unit=`day, then it would equal one day.